#!/usr/bin/env python3

SAMPLES         = ['ATACrep3', 'ATACrep4']
READ_PAIRS      = ['R1', 'R2']
RAW_READS_PATHS = expand('data/{sample}_{read}.fastq.gz', sample=SAMPLES, read=READ_PAIRS)
SAMPLE_NAME     = expand('{sample}_{read_pair}', sample=SAMPLES, read_pair=READ_PAIRS)
TRIM_OUT        = ['r1p', 'r1u', 'r2p', 'r2u']
ALL_NUCLEOSOMES = ['all', 'nfr', 'mono', 'di']
NUCLEOSOMES     = ['nfr', 'mono', 'di']

### Function for generating list of files for rule all
def get_final_output():
    final_list = []

    ### FASTQC OUTPUT
    final_list.append(expand('results/qc/fastqc/{sample}_fastqc.{ext}',
        sample=SAMPLE_NAME,
        ext=['html', 'zip']))


    ### TRIMMOMATIC OUTPUT not needed
    #final_list.append(expand('results/raw_reads_trimmed/{sample}_{trim_out}.fastq.gz', 
    #    sample=SAMPLES,
    #    trim_out=TRIM_OUT))

    ### BOWTIE2 BUILD not needed
    #final_list.append('logs/bowtie2_build.done')
    
    ### BOWTIE2 ALIGNMENT OUTPUTS (requires trimmomatic output) not needed
    #final_list.append(expand('results/bowtie2_alignments/{sample}.bam',
    #    sample=SAMPLES))

    ### SAMTOOLS FLAGSTAT not needed
    #final_list.append(expand('results/qc/flagstats/{sample}.stats',
    #    sample=SAMPLES))

    ### MultiQC Output (requires part of fastqc and all of samtools flagstat output)
    final_list.append('results/qc/multiqc_report.html')

    ### Samtools noMT Output
    #final_list.append(expand('results/samtools/{sample}_noMT.bam',
    #    sample=SAMPLES))

    ### Samtools Index output
    #final_list.append(expand('results/samtools/{sample}_noMT_sorted.bam.bai',
    #    sample=SAMPLES))

    ### DeepTools alignmentSieve
    #final_list.append(expand('results/deeptools/{sample}_{nucleosome}_sieved.bam',
    #    sample=SAMPLES, nucleosome=NUCLEOSOMES))

    ### Deeptools bam sorting and indexing
    #final_list.append(expand('results/deeptools/{sample}_{nucleosome}_sieved_sorted.bam.bai',
    #    sample=SAMPLES, nucleosome=ALL_NUCLEOSOMES))

    ### ATACseqQC output
    final_list.append(expand('results/{sample}_frag_size_dist.png',
        sample=SAMPLES))

    ### MACS3 hmmratac Output
    #final_list.append(expand('logs/macs3_{sample}.done',
    #    sample=SAMPLES))

    ### MACS3 callpeaks output
    #final_list.append(expand('results/macs3/{sample}_all_peaks.narrowPeak',
    #    sample=SAMPLES))

    ### Bedtools Intersect output
    #final_list.append('results/bedtools/ATACpeaks_all.bed')

    ### Bedtools filter blacklist output
    #final_list.append('results/bedtools/ATACpeaks_all_noBlacklist.bed')

    ### Homer annotated peaks output
    #final_list.append('results/homer/ATACpeaks_all_noBlacklist_annotated.bed')

    ### Homer find motifs output
    final_list.append('logs/homer_motifs.done')

    ### Deeptools bam compare output
    #final_list.append(expand('results/plot_profile/ATACreps_{nucleosome}_sieved_sorted.bw',
    #    nucleosome=NUCLEOSOMES))

    ### Deeptools compute matrix output
    #final_list.append('results/plot_profile/ATACreps_nucleosomes_matrix.gz')

    ### Deeptools plot Profile png output
    final_list.append('results/plot_profile/nucleosome_signal_coverage.png')

    ### Regions Pie Chart
    final_list.append('results/regions_pie_chart.png')


    return final_list



### Function for generating list of qc reports to concatenate
#   with MultiQC
def get_qc_reports():
    qc_reports = []

    ### FastQC reports
    qc_reports.append(expand('results/qc/fastqc/{sample}_fastqc.zip',
        sample=SAMPLE_NAME))

    ### Samtools Flagstat reports
    qc_reports.append(expand('results/qc/flagstats/{sample}.stats',
        sample=SAMPLES))
        
    ### Samtools Flagstat on noMT alignments reports
    qc_reports.append(expand('results/qc/flagstats/{sample}_noMT.stats',
        sample=SAMPLES))

    return qc_reports


def get_replicates():
    return SAMPLES
